internal/services/environment_settings/resources_environment_settings.go

```diff
@@ -118,6 +118,14 @@ func (r *EnvironmentSettingsResource) Schema(ctx context.Context, req resource.S
 									boolplanmodifier.UseStateForUnknown(),
 								},
 							},
+							"log_retention_period_in_days": schema.Int64Attribute{
+								Description:         "Is audit retention period v2",
+								MarkdownDescription: "Is audit retention period v2",
+								Optional:            true, Computed: true,
+								PlanModifiers: []planmodifier.Int64{
+									int64planmodifier.UseStateForUnknown(),
```

Comment:
There should be a validation for the input given that the allowed range is between 31 and 99999. I leave it to @mattdot is we should allow to go below 31.

Reply:
Curious how the investigation went on boundary values like int max, 0 , 30, -1, int min, etc

Reply:
All the values are accepted, even if don't make sense, for example here we test with -2:
Apply complete! Resources: 2 added, 0 changed, 0 destroyed.

Outputs:

all_settings = {
  "audit_and_logs" = {
    "audit_settings" = {
      "is_audit_enabled" = true
      "is_read_audit_enabled" = true
      "is_user_access_audit_enabled" = true
      "log_retention_period_in_days" = -2
    }

We add validators.